### PR TITLE
angular - Tanner and Daniel

### PR DESCRIPTION
**Pull request by danielcossins11**

### CODE REVIEW

File: front-end-notes/angular.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+});
+ 
+.controller('MyController', function (movie) {
+  expect(movie.title).toEqual('The Matrix');
+});
+```
+
+###Factory
+
+A factory is an injectable function. A factory is a lot like a service in the sense that it is a singleton and dependencies can be specified in the function. The difference between a factory and a service is that a factory injects a plain function so AngularJS will call the function and a service injects a constructor. A constructor creates a new object so new is called on a service and with a factory you can let the function return anything you want. As you will see later on, a factory is a provider with only a $get method.
```
Comment (NewEvolution) [inline front-end-notes/angular.md]: A factory is actually type of service

File: front-end-notes/angular.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+###Model
+Model is strictly data. It does NOT depend on the controller or the view.
+
+###View
+The view is an interface to view and modify the data.
+
+###Controller
+Controllers are operations that can be performed on the data.
+
```
Comment (sethdorris) [inline front-end-notes/angular.md]: Add information on directives?

**Comment by alwadejr** [comment]
I would love to see some information on directives.

File: front-end-notes/angular.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+### Partials
+
+Partials are a way to link the view (html) to the model and controller using two-way data-binding.
+####Two-way data-binding just means that:
+1. When properties in the model get updated, so does the UI.
+2. When UI elements get updated, the changes get propagated back to the model.
+
+##Sources
+https://docs.angularjs.org/api?PHPSESSID=cae8e98e7ca559b4605d75c813b358ee
+http://www.tomdalling.com/blog/software-design/model-view-controller-explained/
```
Comment (alwadejr) [inline front-end-notes/angular.md]: Two of the links were on the same line. You may want to break them up for easier readability. Suggestion would be using * for listing items in markdown.

File: front-end-notes/angular.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
+#AngularJS
+Angular implements a MVC design pattern, which stands for Model View Controller.
+
+###Model
+Model is strictly data. It does NOT depend on the controller or the view.
+
+###View
+The view is an interface to view and modify the data.
+
+###Controller
```
Comment (BrendonPierson) [inline front-end-notes/angular.md]: Maybe some info on controllerAs and this?

File: front-end-notes/angular.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+Angular implements a MVC design pattern, which stands for Model View Controller.
+
+###Model
+Model is strictly data. It does NOT depend on the controller or the view.
+
+###View
+The view is an interface to view and modify the data.
+
+###Controller
+Controllers are operations that can be performed on the data.
```
Comment (sqfish) [inline front-end-notes/angular.md]: A more rigorous definition of controllers might be helpful, particularly describing what is appropriate to put in a controller and what should be reserved for directives and services, etc.

File: front-end-notes/angular.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+```
+
+### Partials
+
+Partials are a way to link the view (html) to the model and controller using two-way data-binding.
+####Two-way data-binding just means that:
+1. When properties in the model get updated, so does the UI.
+2. When UI elements get updated, the changes get propagated back to the model.
+
+##Sources
```
Comment (sqfish) [inline front-end-notes/angular.md]: This is another resource/style guide that I found useful:
https://github.com/johnpapa/angular-styleguide

File: front-end-notes/angular.md (hunk 2 above)
Comment (BrendonPierson) [inline front-end-notes/angular.md]: Directives are the foundation of Angular.

**Pull request merged**